lms/djangoapps/courseware/courses.py

```diff
@@ -383,7 +383,8 @@ def sort_by_start_date(courses):
     """
     Returns a list of courses sorted by their start date, latest first.
     """
-    courses = sorted(courses, key=lambda course: (course.start is None, course.start), reverse=False)
+    courses = sorted(courses, key=lambda course: (
+        course.has_ended(), course.start is None, course.start), reverse=False)
```

Comment:
Nit: This line-break formatting is a little strange, I'd prefer something along the lines of:
```
    courses = sorted(
        courses,
        key=lambda course: (course.has_ended(), course.start is None, course.start),
        reverse=False
    )
```